test/extended/router/gatewayapicontroller.go

```diff
@@ -334,9 +330,10 @@ func createAndCheckGateway(oc *exutil.CLI, gwname, gwclassname, domain string) (
 func checkGatewayStatus(oc *exutil.CLI, gwname, ingressNameSpace string) (*gatewayapiv1.Gateway, error) {
 	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
 	gateway := &gatewayapiv1.Gateway{}
+	var errGwStatus error
```

Comment:
Why this change?

I would rather do this to make it similar to the function you add (`errGwStatus` -> `err`, more detailed log message:)
```
	gateway, err := gwapiClient.GatewayV1().Gateways(ingressNameSpace).Get(context, gwname, metav1.GetOptions{})
	if err != nil {
		e2e.Logf("Failed to get gateway %q: %v, retrying...", gwname, err)
		return false, nil
	}
```

Reply:
done